### PR TITLE
Adding Pet Mod of the Season, [Gen 3] Hoenn Gaiden

### PR DESCRIPTION
**Pull request by Yoshiblaze**
Hello!

This PR is for adding Hoenn Gaiden to the server as the (long, LONG overdue) Pet Mod of the Season. Another PR for the same purpose was made earlier by Yak Attack, but this one has many notable differences, mainly being more concise and lacking some unnecessary files. 

Any questions can be directed towards Rezzo, Yak Attack, or myself. 

https://www.smogon.com/forums/threads/hoenn-gaiden-pet-mod-of-the-season.3714737/
https://www.smogon.com/forums/threads/hoenn-gaiden-is-gen-9s-first-pet-mod-of-the-season.3716671/ (Decided all the way back in February)

### CODE REVIEW

File: config/formats.ts [hunk 1]
```diff
@@ -540,6 +540,17 @@ export const Formats: FormatList = [
 	{
 		section: "Pet Mods",
 	},
+	{
+		name: "[Gen 3] Hoenn Gaiden",
+		desc: ["<b>Hoenn Gaiden</b>: A Gen 3 pet mod that aims to devamp Gen 4-8 Pokemon, moves and items into the Gen 3 mechanics."],
```
Comment (KrisXV) [inline config/formats.ts:545]:
```suggestion
		desc: `A Gen 3 Pet Mod that aims to devamp Gen 4-8 Pok&eacute;mon, moves, and items into the Gen 3 mechanics.`,
```

File: config/formats.ts [hunk 2]
```diff
@@ -540,6 +540,17 @@ export const Formats: FormatList = [
...
+		threads: [
+			`&bullet; <a href="https://www.smogon.com/forums/threads/hoenn-gaiden-pet-mod-of-the-season.3714737/">Hoenn Gaiden on Smogon Forums</a>`,
+		],
+		mod: 'gen3hoenngaiden',
```
Comment (KrisXV) [inline config/formats.ts:549]:
```suggestion
		],

		mod: 'gen3hoenngaiden',
```

**Review by KrisXV — CHANGES_REQUESTED** [review]
please fix all the build errors

**Comment by Yoshiblaze** [comment]
> please fix all the build errors

Will do, but I'm not quite sure what the specific errors are. I assume they come from the learnsets.ts file since that's the one that was ignored by ESLint, but since it was ignored I'm not sure what to fix in there and I don't see any errors when I run it locally. Is there another way to see what specific errors there are?

**Comment by singiamtel** [comment]
Running ``npm run full-test`` gives me around 10 typing errors

**Comment by Yoshiblaze** [comment]
> Running `npm run full-test` gives me around 10 typing errors

Thank you, after doing this a few times I've now cleared all the errors.

**Comment by Yoshiblaze** [comment]
Had to make some minor changes here and there over the last few weeks, but this should be good to go for September now with the build errors fixed, let me know if I missed anything

File: data/mods/gen3hoenngaiden/pokedex.ts [hunk 3]
```diff
@@ -0,0 +1,1098 @@
...
+	silvally: {
+		inherit: true,
+		otherFormes: ["Silvally-Bug", "Silvally-Dark", "Silvally-Dragon", "Silvally-Electric", "Silvally-Fighting",
+						  "Silvally-Fire", "Silvally-Flying", "Silvally-Ghost", "Silvally-Grass", "Silvally-Ground", "Silvally-Ice",
+						  "Silvally-Poison", "Silvally-Psychic", "Silvally-Rock", "Silvally-Steel", "Silvally-Water"],
+		formeOrder: [
+			"Silvally", "Silvally-Fighting", "Silvally-Flying", "Silvally-Poison", "Silvally-Ground", "Silvally-Rock",
+			"Silvally-Bug", "Silvally-Ghost", "Silvally-Steel", "Silvally-Fire", "Silvally-Water", "Silvally-Grass",
+			"Silvally-Electric", "Silvally-Psychic", "Silvally-Ice", "Silvally-Dragon", "Silvally-Dark",
+		],
```
Comment (KrisXV) [inline data/mods/gen3hoenngaiden/pokedex.ts:48]:
```suggestion
		otherFormes: ['Silvally-Bug', 'Silvally-Dark', 'Silvally-Dragon', 'Silvally-Electric', 'Silvally-Fighting', 'Silvally-Fire', 'Silvally-Flying', 'Silvally-Ghost', 'Silvally-Grass', 'Silvally-Ground', 'Silvally-Ice', 'Silvally-Poison', 'Silvally-Psychic', 'Silvally-Rock', 'Silvally-Steel', 'Silvally-Water'],
		formeOrder: ['Silvally', 'Silvally-Fighting', 'Silvally-Flying', 'Silvally-Poison', 'Silvally-Ground', 'Silvally-Rock', 'Silvally-Bug', 'Silvally-Ghost', 'Silvally-Steel', 'Silvally-Fire', 'Silvally-Water', 'Silvally-Grass', 'Silvally-Electric', 'Silvally-Psychic', 'Silvally-Ice', 'Silvally-Dragon', 'Silvally-Dark'],
```

**Review by KrisXV — APPROVED** [review]

**Comment by KrisXV** [comment]
here's to hoping this doesnt explode

**Pull request merged**